File: UVA10193.cpp
    #include <bits/stdc++.h>
    using namespace std;
    int main()
    {
        int total, num1, num2, num, num_b,num_a, gcd, tmp;
        string str1, str2;
        bool flag;
        cin >> total;
        for (int time = 0; time < total; time++)
        {
            cin >> str1 >> str2;
            gcd = 1;
            flag = true;
            // cout << str1 << " " << str2 << endl;
            num1 = 0; 
            num = 1;
            num2 = 0; 
            for (int i = str1.length() - 1; i >= 0; i--)
            {
                // cout << str1[i] - '0';
                if (str1[i] - '0' == 1)
                {
                    // cout << num;
                    num1 += num;
                }
                num *= 2;
            }
            num = 1;
            for (int i = str2.length() - 1; i >= 0; i--)
            {
                // cout << str2[i] - '0';
                if (str2[i] - '0' == 1)
                {
                    num2 += num;
                }
                num *= 2;
            }
            // cout << num1 << " " << num2 << endl;
            num_a = max(num1, num2);
            num_b = min(num1, num2);
            // cout << num_a << " 777  " << num_b << endl;
            while(num_b > 0)
            {
                tmp = num_a % num_b;
                num_a = num_b;
                num_b = tmp;
            }
            if (num_a != 1)
            {
                cout << "Pair #" << time + 1 << ": All you need is love!" << endl;
            }
            else
            {
                cout << "Pair #" << time + 1 << ": Love is not all you need!" << endl;
            }
        }
        return 0;
    }

File: UVA10193.py
total = int(input())
for time in range(total):
    print(time)
    str1 = input()
    str2 = input()
    num1 = 0
    num2 = 0
    num = 1
    
    for i in range(len(str1) - 1,-1,-1):
        if str1[i] == '1':
            num1 += num
        num *= 2
        # print(num1)
        
    num = 1
    
    for i in range(len(str2) - 1,-1,-1):
        if str2[i] == '1':
            num2 += num
        num *= 2
        
    min_num = min(num1, num2)
    
    gcd = 1
    flag = True
    
    for i in range(2, min_num + 1):
        if num1 % i == 0 and num2 % i == 0:
            if gcd != 1:
                flag = False
                break
            gcd = i
            
    # print(num1,num2,gcd)
    if (flag and gcd != 1):
        print("Pair #{}: All you need is love!".format(time + 1))
    else:
        print("Pair #{}: Love is not all you need!".format(time + 1))
